### ISSUE TITLE
OAuth with Salesforce and Yahoo fails

### ISSUE THREAD

**Issue by jordanbtucker** [feathersjs/feathers#1614]
### Steps to reproduce

Setup OAuth according to the guide, but substitute GitHub for Salesforce or Yahoo.

Attempt to authenticate via Salesforce or Yahoo.

http://localhost:3030/oauth/salesforce

http://localhost:3030/oauth/yahoo

### Expected behavior
The console should log a user being created.

### Actual behavior
An error occurs because the `profile_url` in `grant-profile` for salesforce is wrong and the one for yahoo is empty.

I've already created PRs to fix this on `grant-profile` at https://github.com/simov/grant-profile/pull/8 and https://github.com/simov/grant-profile/pull/9

### System configuration

Tell us about the applicable parts of your setup.

**Module versions** (especially the part that's not working):
```
"@feathersjs/authentication": "^4.3.5",
"@feathersjs/authentication-oauth": "^4.3.5",
"@feathersjs/errors": "^4.3.4",
"@feathersjs/express": "^4.3.5",
"@feathersjs/feathers": "^4.3.4",
"@feathersjs/socketio": "^4.3.5",
"@feathersjs/authentication-client": "^4.3.5",
"@feathersjs/socketio-client": "^4.3.5",
```

**NodeJS version**:
v10.16.3

**Operating System**:
Windows 10

**Browser Version**:
Chrome 77

**Comment by daffl** [comment]
Perfect. Another alternative would be to override the `getProfile` in the strategy with your own call [as shown in the Facebook Cookbook example](https://docs.feathersjs.com/cookbook/authentication/facebook.html#getting-profile-data).

**Comment by jordanbtucker** [comment]
Oh, I forgot to include the workarounds. You can manually set the `profile_url` in the authentication config. For example.

```json
{
  "authentication": {
    "oauth": {
      "salesforce": {
        "key": "<salesforce oauth key>",
        "secret": "<salesforce oauth secret>",
        "profile_url": "https://login.salesforce.com/services/oauth2/userinfo"
      },
      "yahoo": {
        "key": "<yahoo oauth key>",
        "secret": "<yahoo oauth secret>",
        "profile_url": "https://api.login.yahoo.com/openid/v1/userinfo"
      }
    }
}
```

**Comment by daffl** [comment]
Ah, yes I forgot that's possible now. Meant to update the Facebook example accordingly when I heard about it.

**Comment by simov** [comment]
Merged both PRs and published v0.0.6 with a bunch of other profile URLs that I was able to test. Still quite a few to go, but we're getting there.
